pgmpy/inference/Sampling.py

```diff
@@ -257,10 +257,10 @@ def _get_kernel_from_bayesian_model(self, model):
         self.variables = np.array(model.nodes())
         self.cardinalities = {var: model.get_cpds(var).variable_card for var in self.variables}
 
-        for var in self.variables:
-            other_vars = [v for v in self.variables if var != v]
+        for Var in self.variables:
```

Comment:
if look down below the person who wrote this was using var in a for loop also. Which caused some values to skipped in self.transition_models.